jupyter-client.el

```diff
@@ -977,9 +977,13 @@ to the above explanation."
         (setq had-result t)
         (jupyter-with-display-buffer "display" req
           (jupyter-insert data metadata)
-          (jupyter-display-current-buffer-reuse-window
-           :display-data nil (unless (jupyter-pop-up-frame-p :display-data)
-                               #'display-buffer-below-selected))))
+          (if (> (line-number-at-pos (point-max))
+                 jupyter-eval-short-result-max-lines)
+              (jupyter-display-current-buffer-reuse-window
+               :display-data nil (unless (jupyter-pop-up-frame-p :display-data)
+                                   #'display-buffer-below-selected))
+            (funcall jupyter-eval-short-result-display-function
+                     (buffer-string)))))
```

Comment:
> OK I found the issue with the language specific methods not being called in the display-data callback, see 005f2e3. There should be no reason to modify any of the evaluation functions.

Thanks, now `jupyter-eval-line-or-region` calls the R-specific method as I'd like, and in particular calls `browse-url` on the result when appropriate. However, it also pops up an empty `*jupyter-display*` buffer (it's empty because of `jupyter--delete-script-tags`). So I added this snippet to use `jupyter-eval-short-result-display-function` when the output is short.

Reply:
I don't think having the `jupyter-eval-short-result-display-function` here is the right approach since its intended purpose is to display text results which is why its only used in `jupyter--display-eval-result`. If `data` contains an image to be displayed, `jupyter-insert` will insert a single space into the buffer and use the `display` text property to render the image (see `insert-image`). In this case, `jupyter-eval-short-result-display-function` would be called since there is only one line in the buffer containing a single space when really the buffer in which the image was inserted should be shown.

Since nothing is actually inserted into the buffer when calling `jupyter-browser-url-in-temp-file`, what you can do is to show the `*jupyter-display*` buffer only when `(/= (point-min) (point-max))` since it doesn't make sense to display an empty buffer in the first place. Could you also put a comment to explain why this is being done as well.

Reply:
Good point, it looks terrible when `jupyter-eval-string` returns an image. I've fixed this as per your suggestion.